fdwli3ds/echopulse.py

```diff
@@ -41,20 +67,79 @@ def __init__(self, options, columns):
         """
         super(EchoPulse, self).__init__(options, columns)
         # Resolve data files found in directory
-        directory = options['directory']
-        sources = (source for source in os.listdir(directory)
+        self.basedir = options['directory']
+        sources = (source for source in os.listdir(self.basedir)
                    if subtree_pattern.match(source))
         self.source_dirs = [
-            os.path.realpath(os.path.join(directory, source))
+            os.path.realpath(os.path.join(self.basedir, source))
             for source in sources
-            if os.path.isdir(os.path.join(directory, source))
+            if os.path.isdir(os.path.join(self.basedir, source))
         ]
-        # pcschema.xml must be present in the directory
-        self.pcschema = os.path.join(options['directory'], 'pcschema.xml')
+        # default mapping for coordinates
+        self.new_dimnames = {
+            'range': 'x',
+            'theta': 'y',
+            'phi': 'z'
+        }
+        # get custom mapping given in options
+        varmapping = [
+            opt for opt in options.keys()
+            if opt.startswith('map_')
+        ]
+        for var in varmapping:
+            self.new_dimnames.update({var.strip('map_'): options[var]})
+
+        # get pointcloud structure from the directory tree
+        self.ordered_dims = self.scan_structure()
 
         log_to_postgres('{} echo/pulse directories linked'
                         .format(len(self.source_dirs)))
 
+    @property
+    def pcschema(self):
+        xml = schema_skeleton.format(
+            dimensions='\n'.join([
+                xml_dimension.format(
+                    idx, size,
+                    self.new_dimnames.get(name, name), dtype)
+                for idx, size, name, dtype in self.ordered_dims
```

Comment:
Yes it's intended, the ordering is only used before mapping, because we need it with raw names in the execute function (https://github.com/LI3DS/fdw-li3ds/pull/21/files#diff-b400b3c9fe9649d0ef4dc1afb304e37cR180)